### ISSUE TITLE
after the conv function processed, the conditional query returns incorrrectly

### ISSUE THREAD

**Issue by apollodafoni** [pingcap/tidb#58425]
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)

```
create table lrr_test(`COL1` bit(1));
insert into lrr_test values(0x01);
select col1, conv(col1, 2, 16) from lrr_test where conv(col1, 2, 16) = conv(0x01, 2, 16);
```

### 2. What did you expect to see? (Required)
+------------+-------------------+
| col1       | conv(col1, 2, 16) |
+------------+-------------------+
| 0x01       | 1                 |
+------------+-------------------+

### 3. What did you see instead (Required)
Empty set

### 4. What is your TiDB version? (Required)

Release Version: v7.5.5
Edition: Community
Git Commit Hash: a0a8adb1e31bc1ff6fbb62a5bc4900e344877cd7
Git Branch: HEAD
UTC Build Time: 2024-12-19 07:47:15
GoVersion: go1.21.13
Race Enabled: false
Check Table Before Drop: false
Store: tikv

**Comment by apollodafoni** [comment]
v8.5.0 returns the correct value, and it seems to be fixed in a later version.

**Comment by Defined2014** [comment]
I think it be fixed by https://github.com/pingcap/tidb/pull/53502, if you want fix it in v7.5, you could cherry pick it to v7.5
